remove systemd drop-in unit on raspbian

bamarni opened the pull request:
This PR is related to the raspbian integration on docker machine (https://github.com/docker/machine/pull/3605).

Docker machine relies on a `/etc/systemd/system/docker.service` file with all required flags in ExecStart (tcp socket, tls config, ...) but at the moment it gets overwritten because of the drop-in file, breaking the systemd integration.

Looking at https://docs.docker.com/engine/userguide/storagedriver/overlayfs-driver/, it seems like overlayfs is modern and promising, I was surprised to see it so low in the priority list. Would it make sense to move it one level-up?

(If this PR is not accepted, an alternative is to revert #25684.)

bamarni commented:
cc @DieterReuter

justincormack commented:
I think the two parts of this commit should be separate PRs. Changing the default order of graph drivers should not be in a PR called "updates for overlay on raspbian".

bamarni commented:
@justincormack : ok, I've extracted the priority change to its own PR. As per the change in install.sh, it'd really depend on whether or not the other PR is accepted, we'd have to either keep it as it is now or revert #25684. I've marked this PR as "blocked".

DieterReuter commented:
I would suggest to revert #25684, which should be the safest option w/o any side effects at all, and not changing the graph-driver priorities.

justincormack commented:
Yes, it looks like changing the priority is problematic.

bamarni commented:
So... I've updated the PR :) PTAL

justincormack commented on hack/install.sh:
```diff
@@ -397,9 +397,10 @@ do_install() {
-				( set -x; $sh_c "echo '[Service]\nExecStart=\nExecStart=/usr/bin/dockerd --storage-driver overlay -H fd://' > /etc/systemd/system/docker.service.d/overlay.conf" )
+				# Create Raspbian specific systemd unit file, use overlay by default
+				( set -x; $sh_c "mkdir -p /etc/systemd/system" )
+				( set -x; $sh_c "$curl https://raw.githubusercontent.com/docker/docker/master/contrib/init/systemd/docker.service > /etc/systemd/system/docker.service" )
```
I don't like using `master` here. Surely it should be the current commit?

bamarni replied:
From what I can see this script would install from either the main, test or experimental repo, leading to different docker versions, so I'm not sure if there's a way to get the commit hash at that point?

What about handling this at the debian package level? That would also avoid a dependency on github being up.

justincormack replied:
I would think that the unit file should be in the deb, marked as a config file so the user can override.

bamarni replied:
Agreed. How about this script, should we keep this patch or remove it and let it default to devicemapper on raspbian until the deb is updated?

justincormack replied:
I guess defaulting to devicemapper is ok for now.

bamarni committed:
Remove systemd drop-in file on raspbian

It makes systemd integration less intrusive, notably needed for
docker machine (cf. https://github.com/docker/machine/pull/3605).

The default storage driver would now be devicemapper, until the
deb package is updated.

Signed-off-by: Bilal Amarni <bilal.amarni@gmail.com>

bamarni commented:
@justincormack : I've just updated it.

justincormack commented:
LGTM

unclejack commented:
LGTM
